cloud-nio/cloud-nio-impl-drs/src/main/scala/cloud/nio/impl/drs/DrsCloudNioFileProvider.scala

```diff
@@ -0,0 +1,61 @@
+package cloud.nio.impl.drs
+
+import java.nio.channels.{ReadableByteChannel, WritableByteChannel}
+
+import cloud.nio.spi.{CloudNioFileList, CloudNioFileProvider, CloudNioRegularFileAttributes}
+import com.typesafe.config.Config
+import org.apache.http.HttpStatus
+import org.apache.http.client.methods.CloseableHttpResponse
+import org.apache.http.impl.client.{CloseableHttpClient, HttpClientBuilder}
+
+import scala.util.Try
+
+
+class DrsCloudNioFileProvider(config: Config, scheme: String, drsPathResolver: DrsPathResolver) extends CloudNioFileProvider {
+
+  private def getDrsPath(cloudHost: String, cloudPath: String): String = s"$scheme://$cloudHost/$cloudPath"
+
+  private def checkIfPathExistsThroughMartha(drsPath: String): Boolean = {
+    val httpClient: CloseableHttpClient = HttpClientBuilder.create().build()
+
+    try {
+      val marthaResponse: CloseableHttpResponse = drsPathResolver.makeHttpRequestToMartha(drsPath, httpClient)
+
+      try {
+        marthaResponse.getStatusLine.getStatusCode == HttpStatus.SC_OK
+      } finally {
+        Try(marthaResponse.close())
+        ()
+      }
+    } finally {
+      Try(httpClient.close())
+      ()
+    }
+  }
+
+
+  override def existsPath(cloudHost: String, cloudPath: String): Boolean =
+    checkIfPathExistsThroughMartha(getDrsPath(cloudHost, cloudPath))
+
+  //This method is used to check if the given path is a directory or not. Since DRS currently doesn't
+  //support resolving url to directories, it is false
+  override def existsPaths(cloudHost: String, cloudPathPrefix: String): Boolean = false
+
+  override def listObjects(cloudHost: String, cloudPathPrefix: String, markerOption: Option[String]): CloudNioFileList =
+    throw new UnsupportedOperationException("DRS currently doesn't support listObjects.")
```

Comment:
`UOE` is fine, but so is generating a list-of-one-element `if (existsPath(cloudHost, cloudPathPrefix) == true)`